stl/inc/complex

```diff
@@ -1585,21 +1585,23 @@ _NODISCARD complex<_Ty> log(const complex<_Ty>& _Left) {
 
 // FUNCTION TEMPLATE pow
 template <class _Ty>
-_NODISCARD complex<_Ty> pow(const complex<_Ty>& _Left, int _Right) {
-    complex<_Ty> _Tmp = _Left;
-    auto _Count       = static_cast<unsigned int>(_Right);
+_NODISCARD complex<_Common_float_type_t<_Ty, double>> pow(const complex<_Ty>& _Left, int _Right) {

```

Comment:
In the Standard, I see WG21-N4842 26.4.8 [complex.transcendentals]/20 specifying:
```cpp
template<class T> complex<T> pow(const complex<T>& x, const complex<T>& y);
template<class T> complex<T> pow(const complex<T>& x, const T& y);
template<class T> complex<T> pow(const T& x, const complex<T>& y);
```
which are implemented immediately below here. Then 26.4.9 [cmplx.over]/3 specifies the "sufficient additional overloads" which appear to be implemented by:
https://github.com/microsoft/STL/blob/70e49a01568ea6b64cfc56c6baa8881dc992ff72/stl/inc/complex#L1801-L1817

So it appears that this `pow(const complex<_Ty>& _Left, int _Right)` overload is selected only when the RHS is exactly `int`. If it's anything else (e.g. `unsigned int`, `long`), then `pow(const complex<_Ty1>& _Left, const _Ty2& _Right)` will be a better match.

Therefore, it looks like this `pow(const complex<_Ty>& _Left, int _Right)` is a performance optimization. But if that's the case, should it be generalized to accept any integer type as the RHS, and then disambiguated with `pow(const complex<_Ty1>& _Left, const _Ty2& _Right)` (e.g. via SFINAE)? The `static_cast<unsigned int>` will also need to change.

Reply:
It's simular as https://github.com/microsoft/STL/blob/5eb6c98bba24d7a96bc1f4f91833e0369db57dcf/stl/inc/complex#L1822
Moreover, this function is:
* Has SFINAE
* No static_cast<unsigned int>(_Right)

Reply:
I'm not sure yet, but it seems to me that we can leave one of these functions altogether. I'll check more closely.

Reply:
Good catch, that was totally broken (and missed for years because the "Gaussian integer" support is a non-Standard extension and extremely obscure).